Source/WebCore/platform/graphics/gbm/GBMBufferSwapchain.cpp

```diff
@@ -43,11 +43,49 @@ GBMBufferSwapchain::GBMBufferSwapchain(BufferSwapchainSize size)
 
 GBMBufferSwapchain::~GBMBufferSwapchain() = default;
 
+static inline bool isBufferFormatSupported(const DMABufFormat& format)
+{
+    switch (format.fourcc) {
+    case DMABufFormat::FourCC::XRGB8888:
+    case DMABufFormat::FourCC::XBGR8888:
+    case DMABufFormat::FourCC::RGBX8888:
+    case DMABufFormat::FourCC::BGRX8888:
+    case DMABufFormat::FourCC::ARGB8888:
+    case DMABufFormat::FourCC::ABGR8888:
+    case DMABufFormat::FourCC::RGBA8888:
+    case DMABufFormat::FourCC::BGRA8888:
+    case DMABufFormat::FourCC::I420:
+    case DMABufFormat::FourCC::YV12:
+    case DMABufFormat::FourCC::A420:
+    case DMABufFormat::FourCC::NV12:
+    case DMABufFormat::FourCC::NV21:
+    case DMABufFormat::FourCC::YUY2:
+    case DMABufFormat::FourCC::YVYU:
+    case DMABufFormat::FourCC::UYVY:
+    case DMABufFormat::FourCC::VYUY:
+    case DMABufFormat::FourCC::VUYA:
+    case DMABufFormat::FourCC::AYUV:
+    case DMABufFormat::FourCC::Y444:
+    case DMABufFormat::FourCC::Y41B:
+    case DMABufFormat::FourCC::Y42B:
+        return true;
+    default:
+        return false;
+    }
+}
+
 RefPtr<GBMBufferSwapchain::Buffer> GBMBufferSwapchain::getBuffer(const BufferDescription& description)
 {
     auto* device = GBMDevice::singleton().device();
-    if (!device)
+    if (!device) {
+        WTFLogAlways("Failed to get GBM buffer from swap chain: no GBM device found");
         return nullptr;
+    }
+
+    if (!isBufferFormatSupported(description.format)) {
+        WTFLogAlways("Failed to get GBM buffer from swap chain: unsupported format");
+        return nullptr;
+    }
```

Comment:
This position can unnecessarily leave live buffer objects that would otherwise be destroyed in the next if-clause.